src/imitation/algorithms/preference_comparisons.py

```diff
@@ -946,46 +993,35 @@ def train(
         Returns:
             A dictionary with final metrics such as loss and accuracy
             of the reward model.
-
-        Raises:
-            ValueError: `total_comparisons < self.comparisons_per_iteration`.
         """
         initial_comparisons = int(total_comparisons * self.initial_comparison_frac)
         total_comparisons -= initial_comparisons
-        iterations, extra_comparisons = divmod(
-            total_comparisons,
-            self.comparisons_per_iteration,
-        )
-        if iterations == 0:
-            raise ValueError(
-                f"total_comparisons={total_comparisons} is less than "
-                f"comparisons_per_iteration={self.comparisons_per_iteration}",
-            )
-        timesteps_per_iteration, extra_timesteps = divmod(total_timesteps, iterations)
 
+        # Compute the number of comparisons to request at each iteration in advance.
+        vec_schedule = np.vectorize(self.query_schedule)
```

Comment:
We could make query schedule ndarray->ndarray rather than float->float. No real performance difference as only call it once, and the float->float is more general. But is there a case where we couldn't easily write a vector implementation? (I think all the current ones would "just work" with ndarray)

Reply:
I thought about doing `np.ndarray -> np.ndarray`, and actually coded it up that way at first, but I decided to change it to `float -> float` for two main reasons: 1) `torch.optim.lr_scheduler.LambdaLR` uses `float -> float` and I wanted to err on the side of being similar to existing "schedule" APIs, and 2) the docstring I wrote up for the `np.ndarray` approach seemed wordier and harder to follow than the current one.

I can't think of a case where `np.ndarray -> np.ndarray` wouldn't also work.

Reply:
Keeping it the same as the existing schedule API is a good reason, we do this in `BetaSchedule` in DAgger as well and `stable_baselines3.Schedule`, so I think I'm overall convinced.